Automattic/wp-calypso
# Congrats: Refactored domain transfer to user congrats page

Related to https://github.com/Automattic/dotcom-forge/issues/5123

## Proposed Changes

This PR refactors the Domain Transferred to User Congrats Page adapt the new approach for unifying the congratulation pages using the `<ThankYouV2>` component. See pb5gDS-38y-p2#comment-4475 for details.

## Testing Instructions



* Run the https://github.com/Automattic/wp-calypso/pull/86201#issuecomment-1884056535 or checkout this branch and run it locally.
* Hit the URL `/checkout/domain-transfer-to-any-user/thank-you/{domain-name}` (any domain name works, e.g. `/checkout/domain-transfer-to-any-user/thank-you/awongdomainonly3.com`).
* Visit the production page and assert that the pages render the same content, with the following changes:
  * ~The confetti animation removed.~ [**UPDATE**: We are keeping the animation after all p9Jlb4-7Hx-p2#comment-10513]
  * Copy changed of `Master the domain essentials` to `Learn the domain essentials` (see https://github.com/Automattic/wp-calypso/pull/86186#discussion_r1448627495)
  * Padding is different from what we currently have in production for the Transfer to User Page. However, it is consistent with the Domain Only Page, since that we're using the exact same component to render these pages. So I'm incline to move forward with the updated padding. We can adjust the padding universally for all Congrats Page if needed as a clean up task (FYI @JanaMW27)

| Current Transfer to User Page | Updated Transfer to User Page | Current Domain Only Page |
| ------- | ------ | ------ |
|<img width="1266" alt="Screen Shot 2024-01-11 at 2 46 26 PM" src="https://github.com/Automattic/wp-calypso/assets/104910361/39acd3ed-4707-422c-b725-505870826a91">|<img width="1265" alt="Screen Shot 2024-01-11 at 2 46 07 PM" src="https://github.com/Automattic/wp-calypso/assets/104910361/40fb1457-d04f-4188-8699-7ebf914417b4">|<img width="1264" alt="Screen Shot 2024-01-11 at 2 45 49 PM" src="https://github.com/Automattic/wp-calypso/assets/104910361/8ab66e21-c702-4d0d-824e-8227d4703a80">|
